dotCMS/src/main/java/com/dotcms/rest/RoleResource.java

```diff
@@ -133,12 +102,47 @@ public Response loadChildren(@Context HttpServletRequest request, @PathParam("pa
 						jsonChildren.add(jsonRoleChildObject);
 					}
 				}
-			}
-			//In order to add a JsonArray to a JsonObject
-			//we need to specify that is an object (API bug)
-			jsonRoleObject.put("children", (Object)jsonChildren);
+				//In order to add a JsonArray to a JsonObject
+				//we need to specify that is an object (API bug)
+				jsonRoleObject.put("children", (Object) jsonChildren);
+				jsonRoles.add(jsonRoleObject);
+
+				return responseResource.response(jsonRoles.toString(), cc);
+
+			} else {  // Loads Children Roles of given Role ID
+				Role role = roleAPI.loadRoleById(roleId);
+
+				JSONObject jsonRoleObject = new JSONObject();
+				jsonRoleObject.put("id", role.getId());
+				jsonRoleObject.put("name", role.getName());
+				jsonRoleObject.put("locked", role.isLocked());
+
+				JSONArray jsonChildren = new JSONArray();
+
+				List<String> children = role.getRoleChildren();
+				if (children != null) {
+					for (String childId : children) {
+						Role r = roleAPI.loadRoleById(childId);
+						if (!workflowRoles.contains(r)) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)